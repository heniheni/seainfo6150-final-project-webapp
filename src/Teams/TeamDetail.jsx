import React from "react";
import styles from "./TeamDetail.module.css";
import { Link } from "react-router-dom";
import bcci from "../images/bcci.png";

const TeamDetail = (props) => {
  let SquadList;

  if (props.team.squad.length) {
    SquadList = (
      <div className={styles.listingMainContainer}>
        {props.team.squad.map((playername) => (
          <div className={styles.listingName} key={playername}>{playername}</div>
        ))}
      </div>
    );
  } else {
    SquadList = <div>You have no data!</div>;
  }
  //const image1 = require(props.team.teamLogoUrl);
  return (
    <div>
      <Link className={styles.stylingBreadCrumb} to="/">Home{' >'}</Link><Link className={styles.stylingBreadCrumb} to="/Teams">Teams{' >'}</Link><Link className={styles.stylingBreadCrumb} to={props.team.teamReference}>{props.team.tag}</Link>
      <div className={styles.mainContainer}>
        <img
          className={styles.logoStyling}
          //src={bcci}
          src={props.team.teamLogoUrl}
          alt="Board of Control for Cricket in India"
        />
        <h4 className={styles.teamNameStyling}>{props.team.teamName}</h4>

        <div className={styles.teamDescription}>
          <div className={styles.teamInfo}>
            <div className={styles.contentOne}>
              <div className={styles.subContainer}>
                <small className={styles.fields}>Owner</small>
                <small className={styles.valueStyling}>
                  {props.team.owner}
                </small>
              </div>

              <div className={styles.subContainer}>
                <small className={styles.fields}>Coach</small>
                <small className={styles.valueStyling}>
                  {props.team.coach}
                </small>
              </div>
            </div>

            <div className={styles.contentTwo}>
              <div className={styles.subContainer}>
                <small className={styles.fields}>Venue</small>
                <small className={styles.valueStyling}>
                  {props.team.venue}
                </small>
              </div>

              <div className={styles.subContainer}>
                <small className={styles.fields}>Captain</small>
                <small className={styles.valueStyling}>
                  {props.team.captain}
                </small>
              </div>
            </div>
          </div>

          <div className={styles.contentThree}>
            <div>
              <p>{props.team.teamDescription}</p>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div>
          <h2 className={styles.stylingSquadHeader}>Squad</h2>
        </div>

        <div>
          <ul>
            <li>{SquadList}</li>
          </ul>
        </div>
      </div>
    </div>
  );
};

export default TeamDetail;
