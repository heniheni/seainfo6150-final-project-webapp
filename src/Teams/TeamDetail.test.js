import React from "react";
import TeamDetail from "./TeamDetail.jsx";

describe("TeamDetail tests", () => {
    
    var team = '{"tag": "CSK","teamName": "Chennai Super Kings","teamReference": "chennai-super-kings","owner": "Chennai Super Kings Cricket Ltd","coach": "Stephen Fleming","teamDescription": "Text","captain": "MS Dhoni","venue": "M. A. Chidambaram Stadium","teamLogoUrl": "https://upload.wikimedia.org/wikipedia/en/thumb/2/2b/Chennai_Super_Kings_Logo.svg/596px-Chennai_Super_Kings_Logo.svg.png","tournmentWon": "2010, 2011, 2018","squad":  ["MS Dhoni (c/wk)","Faf du Plessis","Ruturaj Gaikwad","Suresh Raina", "Ambati Rayudu","N. Jagadeesan (wk)","Robin Uthappa","Ravindra Jadeja","Sam Curran","Dwayne Bravo","Karn Sharma","R. Sai Kishore","Mitchell Santner","Imran Tahir","Deepak Chahar","Shardul Thakur","Lungi Ngidi","KM Asif","Moeen Ali","K Gowtham","Cheteshwar Pujara","M Harisankar Reddy","K. Bhagath Varma","C Hari Nishanth","Jason Behrendorff"]}';

    var json = JSON.parse(team);
    it("renders correctly", () => {
      const { container } = render(<TeamDetail team={json} />);
      expect(container).toMatchSnapshot();
    });

});