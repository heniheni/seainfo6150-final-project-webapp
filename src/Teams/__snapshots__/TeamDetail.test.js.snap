// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamDetail tests renders correctly 1`] = `
<div>
  <div>
    <a
      class="stylingBreadCrumb"
      href="/"
    >
      Home 
       &gt; 
    </a>
    <a
      class="stylingBreadCrumb"
      href="/Teams"
    >
       Teams
       &gt; 
    </a>
    <a
      class="stylingBreadCrumb"
      href="/chennai-super-kings"
    >
      CSK
    </a>
    <div
      class="mainContainer"
    >
      <img
        alt="Board of Control for Cricket in India"
        class="logoStyling"
        src="https://upload.wikimedia.org/wikipedia/en/thumb/2/2b/Chennai_Super_Kings_Logo.svg/596px-Chennai_Super_Kings_Logo.svg.png"
      />
      <h4
        class="teamNameStyling"
      >
        Chennai Super Kings
      </h4>
      <div
        class="teamDescription"
      >
        <div
          class="teamInfo"
        >
          <div
            class="contentOne"
          >
            <div
              class="subContainer"
            >
              <small
                class="fields"
              >
                Owner
              </small>
              <small
                class="valueStyling"
              >
                Chennai Super Kings Cricket Ltd
              </small>
            </div>
            <div
              class="subContainer"
            >
              <small
                class="fields"
              >
                Coach
              </small>
              <small
                class="valueStyling"
              >
                Stephen Fleming
              </small>
            </div>
          </div>
          <div
            class="contentTwo"
          >
            <div
              class="subContainer"
            >
              <small
                class="fields"
              >
                Venue
              </small>
              <small
                class="valueStyling"
              >
                M. A. Chidambaram Stadium
              </small>
            </div>
            <div
              class="subContainer"
            >
              <small
                class="fields"
              >
                Captain
              </small>
              <small
                class="valueStyling"
              >
                MS Dhoni
              </small>
            </div>
          </div>
        </div>
        <div
          class="contentThree"
        >
          <div>
            <p>
              Text
            </p>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div>
        <h2
          class="stylingSquadHeader"
        >
          Squad
        </h2>
      </div>
      <div>
        <ul>
          <li>
            <div
              class="listingMainContainer"
            >
              <div
                class="listingName"
              >
                MS Dhoni (c/wk)
              </div>
              <div
                class="listingName"
              >
                Faf du Plessis
              </div>
              <div
                class="listingName"
              >
                Ruturaj Gaikwad
              </div>
              <div
                class="listingName"
              >
                Suresh Raina
              </div>
              <div
                class="listingName"
              >
                Ambati Rayudu
              </div>
              <div
                class="listingName"
              >
                N. Jagadeesan (wk)
              </div>
              <div
                class="listingName"
              >
                Robin Uthappa
              </div>
              <div
                class="listingName"
              >
                Ravindra Jadeja
              </div>
              <div
                class="listingName"
              >
                Sam Curran
              </div>
              <div
                class="listingName"
              >
                Dwayne Bravo
              </div>
              <div
                class="listingName"
              >
                Karn Sharma
              </div>
              <div
                class="listingName"
              >
                R. Sai Kishore
              </div>
              <div
                class="listingName"
              >
                Mitchell Santner
              </div>
              <div
                class="listingName"
              >
                Imran Tahir
              </div>
              <div
                class="listingName"
              >
                Deepak Chahar
              </div>
              <div
                class="listingName"
              >
                Shardul Thakur
              </div>
              <div
                class="listingName"
              >
                Lungi Ngidi
              </div>
              <div
                class="listingName"
              >
                KM Asif
              </div>
              <div
                class="listingName"
              >
                Moeen Ali
              </div>
              <div
                class="listingName"
              >
                K Gowtham
              </div>
              <div
                class="listingName"
              >
                Cheteshwar Pujara
              </div>
              <div
                class="listingName"
              >
                M Harisankar Reddy
              </div>
              <div
                class="listingName"
              >
                K. Bhagath Varma
              </div>
              <div
                class="listingName"
              >
                C Hari Nishanth
              </div>
              <div
                class="listingName"
              >
                Jason Behrendorff
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
